### DISCUSSION TITLE
Slider with fixed available values

### DISCUSSION THREAD

**Discussion by ailchenkoDynamo** [radix-ui/primitives#1674] [Help]
Hi. Can I use slider with fixed available values? User can select only 20, 24,22, 53 ... ? Thanks)

**Comment by benoitgrelard** [comment]
Hey @ailchenkoDynamo,

I don't believe this is possible at the moment. Sliders are usually used for continuous values. It seems like if you're after a control to select from discrete values maybe something like `Select` would work better?

Do you mind sharing what your use case is? Perhaps it is valid and we could look at adding support if so?

**Reply by ailchenkoDynamo** [reply]
@benoitgrelard I have slider with trigger value. It is value from 0 to 100, but for different cases I have to disable some values from list
![image](https://user-images.githubusercontent.com/98591714/191254670-98dd35b4-85eb-45a6-9461-7b656588c9b4.png)

**Reply by benoitgrelard** [reply]
Sorry I'm not understanding exactly what this should achieve by looking at the picture.
Can you explain a bit what the functionality should be?
Thanks!

**Reply by ailchenkoDynamo** [reply]
I have slider with trigger value. It is value from 0 to 100, but for different cases I have to disable some values from list for example 
 
first case: available range from 0 to 100
second: available from 0 - 30 and 40-80
third: available value from 30-40

**Reply by benoitgrelard** [reply]
Right so in this case it's rather that a subrange is allowed? Rather than specific values only being allowed like say out of the [0-100] you can only pick 37, 56 and 78. Or Do you have use cases for that too?

**Reply by ailchenkoDynamo** [reply]
@benoitgrelard Pls see this example with material UI slider. It that what I need. https://mui.com/material-ui/react-slider/#restricted-values

**Reply by benoitgrelard** [reply]
I see, still feels strange to me, like you'd be better off with a select, curious of actual use-cases where this is used in real life.
If we can figure out a way to handle both use-cases at once though, why not. Happy for you to take a stab at it.
Ideally without a breaking change in the API would be our preference.

**Comment by ailchenkoDynamo** [comment]
> Perhaps it is valid and we could look at adding support if so?

I can try to do it if you give me approve for it

**Reply by benoitgrelard** [reply]
https://github.com/radix-ui/primitives/discussions/1674#discussioncomment-3690410
